src/utils.php

```diff
@@ -645,3 +649,133 @@ function ensure_dir( $dir ) {
+ *
+ * @param string $dir The directory path to check.
+ *
+ * @return bool True if the mtime has changed, false otherwise.
+ */
+function should_skip_check_based_on_mtime( string $dir ): bool {
+	$lastruntime_file = root( '/.lastruntime' );
+
+	if ( file_exists( $lastruntime_file ) ) {
+		$last_runtime  = (int) file_get_contents( $lastruntime_file );
+		$current_mtime = filemtime( $dir );
+
+		if ( $current_mtime <= $last_runtime ) {
+			return true;  // Skip the check.
+		}
+	}
+
+	return false;
+}
+
+/**
+ * Check the ownership and ID of a given file.
+ *
+ * @param string $file_path   The file path to check.
+ * @param int    $current_uid The current user ID.
+ * @param int    $current_gid The current group ID.
+ * @param int    $invalid_id  The invalid ID to check against.
+ *
+ * @return string|bool Error message or true if all checks pass.
+ */
+function check_file_ownership_and_id( $file_path, $current_uid, $current_gid, $invalid_id ) {
+	$file_owner = fileowner( $file_path );
+	$file_group = filegroup( $file_path );
+
+	// Using indexed placeholders for sprintf.
+	$message_template = "<red>Slic has identified that the file located at %1\$s is assigned a UID/GID of %2\$d.</red>" . PHP_EOL . PHP_EOL . "This is likely a result of a documented Docker bug (refer to: https://github.com/docker/desktop-linux/issues/31)." . PHP_EOL . PHP_EOL . "To address this issue, execute the following command:" . PHP_EOL . "<light_cyan>sudo chown -R {$current_uid}:{$current_gid} %1\$s</light_cyan>" . PHP_EOL;
+
+	// Check if the file owner or group ID is invalid.
+	if ( $file_owner === $invalid_id || $file_group === $invalid_id ) {
+		return colorize( sprintf( $message_template, $file_path, $invalid_id ) );
+	}
+
+	return true;
+}
+
+/**
+ * Recursively check permissions in a directory for UID 10099.
+ *
+ * @ref https://github.com/docker/desktop-linux/issues/31
+ *
+ * @param string $dir The directory path to check.
+ */
+function check_permissions_in_directory_recursively( string $dir ) {
+
+	if ( 'Linux' !== os() ) {
+		return;
+	}
+
+	$current_uid = getmyuid();
+	$current_gid = getmygid();
+	$invalid_id  = 100999;
+
+	try {
+		$directory_iterator = new RecursiveDirectoryIterator( $dir, FilesystemIterator::SKIP_DOTS );
+		$iterator           = new RecursiveIteratorIterator( $directory_iterator,
+		                                                     RecursiveIteratorIterator::SELF_FIRST );
+
+		foreach ( $iterator as $file_info ) {
+			$file_path = $file_info->getPathname();
+
+			$result = check_file_ownership_and_id( $file_path, $current_uid, $current_gid, $invalid_id );
+			if ( $result !== true ) {
+				exit( $result );  // Early exit.
+			}
+		}
+	} catch ( Exception $e ) {
+		exit( "Error: Unable to open directory {$dir}." );  // Early exit.
+	}
+}
+
+/**
+ * Confirm that all given paths have correct file permissions on Linux.
+ *
+ * @param array $paths The paths to start checking from.
```

Comment:
Here and elsewhere: `array` arguments require documentations about what they contain:
```suggestion
 * @param string[] $paths The paths to start checking from.
```
You can also use the `array<string>` notation.